### DISCUSSION TITLE
Why does RSACng.SignHash() have a HashAlgorithmName parameter?

### DISCUSSION THREAD

**Discussion by drauch** [dotnet/runtime#92929] [Q&A] [answered]
Why does `RSACng.SignHash()` have a `HashAlgorithmName` parameter? From my point of view, it makes sense for the sister method `SignData()` but why does `SignHash()` need such a parameter for already hashed data?

Consequently, why does `ECDsaCng.SignHash()` not have such a parameter?

Best regards,
D.R.

**Comment by stephentoub** [comment, marked as answer]
I see you got an answer to asking the same question on stackoverflow:
https://stackoverflow.com/questions/77222021/why-does-rsacng-signhash-have-a-hashalgorithmname-parameter

There are also similar answers elsewhere on stackoverflow, e.g.
https://stackoverflow.com/questions/11843336/why-does-signhash-need-to-know-what-hash-algorithm-was-used

**Comment by drauch** [comment]
Yep, sorry for crossposting.
